environment/src/network/message.rs

```diff
@@ -255,152 +251,149 @@ impl<N: Network, E: Environment> Message<N, E> {
 
     /// Deserializes the given buffer into a message.
     #[inline]
-    pub fn deserialize<R: Read + Seek>(reader: &mut R) -> Result<Self> {
-        // Read the message ID.
-        let id: u16 = bincode::deserialize_from(&mut *reader)?;
-
-        // Helper function to read all the remaining bytes from a reader.
-        let read_to_end = |reader: &mut R| -> Result<Bytes> {
-            let mut data = vec![];
-            reader.read_to_end(&mut data)?;
+    pub fn deserialize(mut bytes: BytesMut) -> Result<Self> {
+        // Make sure there is an id available.
+        if bytes.remaining() < 2 {
+            bail!("Missing message ID");
+        }
 
-            Ok(data.into())
-        };
+        // Read the message ID.
+        let id: u16 = bytes.get_u16_le();
 
         // Deserialize the data field.
         let message = match id {
-            0 => Self::BlockRequest(bincode::deserialize_from(&mut *reader)?, bincode::deserialize_from(&mut *reader)?),
-            1 => Self::BlockResponse(Data::Buffer(read_to_end(&mut *reader)?)),
+            0 => {
+                let mut reader = bytes.reader();
```

Comment:
Is there no way to use `bytes.reader()` directly?

Reply:
`BytesMut::reader` takes `self` by value and it's used twice in this block, so it's necessary to introduce a binding here.